aesara/link/numba/dispatch/random.py

```diff
@@ -342,4 +342,46 @@ def sampler(rng, size, dtype, p):
 
         return (rng, res)
 
-    return sampler
+    return categorical_rv
+
+
+@numba_funcify.register(aer.DirichletRV)
+def numba_funcify_DirichletRV(op, node, **kwargs):
+
+    out_dtype = node.outputs[1].type.numpy_dtype
+    alphas_ndim = node.inputs[3].type.ndim
+    neg_ind_shape_len = -alphas_ndim + 1
+    size_len = int(get_vector_length(node.inputs[1]))
+
+    if alphas_ndim > 1:
+
+        @numba_basic.numba_njit
+        def dirichlet_rv(rng, size, dtype, alphas):
+
+            if size_len > 0:
+                size_tpl = numba_ndarray.to_fixed_tuple(size, size_len)
+                if (
+                    0 < alphas.ndim - 1 <= len(size_tpl)
+                    and size_tpl[neg_ind_shape_len:] != alphas.shape[:-1]
+                ):
+                    raise ValueError("Parameters shape and size do not match.")
```

Comment:
Doesn't the `broadcast_to` below handle these cases?

Reply:
As I recall, this was added to match NumPy's behavior.